Revise documentation on managing dependencies. (#30450)
apache/beam: website/www/site/content/en/documentation/sdks/python-pipeline-dependencies.md

## PyPI Dependencies {#pypi-dependencies}

If your pipeline uses public packages from the [Python Package Index](https://pypi.python.org/), you must make these packages available remotely on the workers.

> For simplest pipelines where a pipeline consists only of a single Python file or a notebook, the easiest way to supply dependencies is to provide a

Reviewer: ```suggestion
For pipelines that consists only of a single Python file or a notebook, the most straightforward way to supply dependencies is to provide a
```

        setuptools.setup(
           name='PACKAGE-NAME',
           version='PACKAGE-VERSION',
           install_requires=[
             # List Python packages your pipeline depends on.
           ],
           packages=setuptools.find_packages(),
        )

2. Structure your project so that the root directory contains the `setup.py` file, the main workflow file, and a directory with the rest of the files, for example:

        root_dir/
          setup.py
          main.py
          my_package/
            my_pipeline_launcher.py
            my_custom_dofns_and_transforms.py
            other_utils_and_helpers.py

    See [Juliaset](https://github.com/apache/beam/tree/master/sdks/python/apache_beam/examples/complete/juliaset) for an example that follows this project structure.

> 3. Install your package in the submission environment, for example via:

Reviewer: ```suggestion
3. Install your package in the submission environment, for example by using the following command:
```

## Non-Python Dependencies or PyPI Dependencies with Non-Python Dependencies {#nonpython}

> If your pipeline uses non-Python packages (e.g. packages that require installation using the `apt install` command), or uses a PyPI package that depends on non-Python dependencies during package installation, we recommend installing them using a (custom container){#custom-containers}.

Reviewer: ```suggestion
If your pipeline uses non-Python packages, such as packages that require installation using the `apt install` command, or uses a PyPI package that depends on non-Python dependencies during package installation, we recommend installing them using a [custom container](#custom-containers).
```

> Alternatively you must perform the following steps.

Reviewer: ```suggestion
Otherwise, you must perform the following steps.
```

1. [Structure your pipeline as a package](#multiple-file-dependencies).

> 2. Add the required installation commands (e.g. the `apt install` commands) for the non-Python dependencies to the list of `CUSTOM_COMMANDS` in your `setup.py` file. See the [Juliaset setup.py](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/complete/juliaset/setup.py) for an example.

Reviewer: ```suggestion
2. Add the required installation commands for the non-Python dependencies, such as the `apt install` commands, to the list of `CUSTOM_COMMANDS` in your `setup.py` file. See the [Juliaset setup.py file](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/complete/juliaset/setup.py) for an example.
```

>     **Note:** You must make sure that these commands are runnable on the remote worker (e.g. if you use `apt`, the remote worker needs `apt` support).

Reviewer: ```suggestion
    **Note:** You must verify that these commands run on the remote worker. For example, if you use `apt`, the remote worker needs `apt` support.
```
